contrib/init.d/pbs_mom.in

```diff
@@ -28,12 +29,30 @@ else
 fi
 
 pidof_pbs_mom() {
-	pid=`pidof pbs_mom`
-	if [ $? -eq 0 ]; then
-		echo $pid
-	else
-		echo "-1"
-	fi
+    pids=`pidof pbs_mom`
+    if [ $? -ne 0 ]; then
+        echo "-1"
```

Comment:
The problem is that there are two kinds of returned 'values'. There is the return code and the output that is produced. Calling v=`function_f` will put the latter in v, the return code in $?.